src/site/content/en/blog/mediastreamtrack-insertable-media-processing/index.md

```diff
@@ -0,0 +1,252 @@
+
+Insertable streams for `MediaStreamTrack` objects conceptually splits the `MediaStreamTrack` into
+two components:
+
+- The `TrackProcessor`, which consumes a `MediaStreamTrack` object's source and generates a stream
+  of media frames ([`VideoFrame`](https://w3c.github.io/webcodecs/#videoframe-interface) or
+  [`AudioFrame`](https://w3c.github.io/webcodecs/#audioframe-interface)). You can think of this as a
+  track sink that is capable of exposing the unencoded frames from the track to a `ReadableStream`
+  and that exposes a control channel for signals going in the opposite direction.
+- The `TrackGenerator`, which consumes a stream of media frames and exposes a `MediaStreamTrack`
+  interface, so that it can be used anywhere a `MediaStreamTrack` is currently attached. It takes
+  video frames as input, and emits control signals that result from subsequent processing.
+
+#### The `MediaStreamTrackProcessor`
+
+A `MediaStreamTrackProcessor` object exposes two properties:
+
+- `readable`: Allows reading the frames flowing through the `MediaStreamTrack`. If the track is a
+  video track, chunks read from `readable` will be `VideoFrame` objects. If the track is an audio
+  track, chunks read from `readable` will produce `AudioFrame` objects.
+- `writableControl`: Allows sending control signals to the track. Control signals are objects of
+  type `MediaStreamTrackSignal`.
+
+#### The `MediaStreamTrackGenerator`
+
+A `MediaStreamTrackGenerator` object likewise exposes two properties:
+
+- `writable`: A `WritableStream` that allows writing media frames to the
+  `MediaStreamTrackGenerator`, which is itself a `MediaStreamTrack`. If the `kind` attribute is
+  `"audio"`, the stream will accept `AudioFrame` objects and fail with any other type. If kind is
+  `"video"`, the stream will accept `VideoFrame` objects and fail with any other type. When a frame
+  is written to `writable`, the frame's `close()` method is automatically invoked, so that its
+  internal resources are no longer accessible from JavaScript.
+- `readableControl`: A `ReadableStream` that allows reading control signals sent from any sinks
+  connected to the `MediaStreamTrackGenerator`. Control signals are objects of type
+  `MediaStreamTrackSignal`.
+
+In the `MediaStream` model, apart from media, which flows from sources to sinks, there are also
+control signals that flow in the opposite direction (i.e., from sinks to sources via the track). A
+`MediaStreamTrackProcessor` is a sink and it allows sending control signals to its track and source
+via its `writableControl` field. A `MediaStreamTrackGenerator` is a track for which a custom source
+can be implemented by writing media frames to its `writable` field. Such a source can receive
+control signals sent by sinks via its `readableControl` field.
+
+### Bringing it all together
+
+The core idea is to create a processing chain as follows:
+
+```bash
+Platform Track → Processor → Transform → Generator → Platform Sinks
```

Comment:
Why not use actual obect/property names?

Reply:
It’s meant to convey the basic mechanics using pseudo-code-like names.